modulemap/module.modulemap

```diff
@@ -0,0 +1,4 @@
+module libtomcrypt [extern_c] {
+    header "../src/headers/tomcrypt.h"
```

Comment:
If libtom/libtommath#518 is really the right way to do stuff, it would also mean that this should be changed to allow using ltc in other modules or not?

Reply:
but it would make sense to support both (C and Swift/ObjC packages depending on this) or not?

Reply:
> but it would make sense to support both (C and Swift/ObjC packages depending on this) or not?

Swift and ObjC can import package as module, module name defined in `module.modulemap`
But C is not support import module, so use `#include <xx>` syntax import symbols.

Reply:
Run `swift build -v`

header search path in log:
```
-I .build/checkouts/libtommath/modulemap
```

Open `Package.swift` in Xcode and build, header search path in log:

```
-I ~/Library/Developer/Xcode/DerivedData/libtomcrypt-bewymzccinmagxaywggcjblpnfww/SourcePackages/checkouts/libtommath/modulemap
```

Reply:
I'm not sure if this is correct or should instead state `publicHeadersPath: "."`

https://github.com/libtom/libtommath/blob/2cec6addaf4acbd8250611ec4cecadf1f4d655ee/Package.swift#L29

Like that you would also not need libtom/libtommath!518

Reply:
> I'm not sure if this is correct or should instead state publicHeadersPath: "."

It is challenged. 

Set `publicHeadersPath: "."` path same as `sources: ["."]` and use `-fmodule-map-file=modulemap/module.modulemap`.

It is not work:

```
$ swift test

....
./libtommath/mtest/mpi.h:128:8: error: conflicting types for 'mp_mul'
mp_err mp_mul(mp_int *a, mp_int *b, mp_int *c);
       ^
./libtommath/tommath.h:353:8: note: previous declaration is here
mp_err mp_mul(const mp_int *a, const mp_int *b, mp_int *c) MP_WUR;
       ^
<unknown>:0: error: too many errors emitted, stopping now
./libtommath/demo/tommath_tests.swift:2:8: error: could not build Objective-C module 'libtommath'
import libtommath
       ^

error: fatalError
```

Reply:
OK, but you shouldn't build `mtest` in the first place ... also let's move this discussion back to libtom/libtommath#518